online-shop/src/app/add-product/add-product.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Supplier } from 'src/supplier';
+import { ProductCategory } from 'src/product-category';
+import { Subscription } from 'rxjs';
+import { ProductCategoryService } from '../services/product-category.service';
+import { SupplierService } from '../services/supplier.service';
+
+@Component({
+    selector: 'app-add-product',
+    templateUrl: './add-product.component.html',
+    styleUrls: ['./add-product.component.scss']
+})
+export class AddProductComponent implements OnInit {
+    addProductForm!: FormGroup;
+    suppliers: Supplier[] = [];
+    productCategories: ProductCategory[] = [];
+    subscription = new Subscription();
+    constructor(
+        private fb: FormBuilder,
+        private dialogRef: MatDialogRef<AddProductComponent>,
+        private categoryService: ProductCategoryService,
+        private supplierService: SupplierService
+    ) {}
+
+    ngOnInit(): void {
+        this.subscription.add(
+            this.categoryService.getAllProductCategories().subscribe((data: ProductCategory[]) => {
+                this.productCategories = data;
+            })
+        );
+        this.subscription.add(
+            this.supplierService.getAllSuppliers().subscribe((data: Supplier[]) => {
+                this.suppliers = data;
+            })
+        );
+        this.addProductForm = this.fb.group({
+            id: '',
+            name: new FormControl('', [Validators.required]),
+            description: new FormControl('', [Validators.required]),
+            price: new FormControl('', [Validators.required]),
+            weight: new FormControl('', [Validators.required]),
+            imgUrl: new FormControl('', [Validators.required]),
+            supplier: new FormControl(this.suppliers, [Validators.required]),
+            productCategory: new FormControl(this.suppliers, [Validators.required])
```

Comment:
this is a happy case, because you are on your local machine. But when it will be deployed/production, the word '.subscribe' doesn't mean that you will have the value on the next line.
Here what I would do is to extract the this.addProductForm = this.fb.group(...) to a function, and call that function inside the getAllSuppliers.subscribe[AFTER ' this.suppliers = data;']